pasqal_cloud/__init__.py

```diff
@@ -131,35 +128,30 @@ def create_batch(
         if configuration:
             req.update({"configuration": configuration.to_dict()})  # type: ignore
 
-        batch_rsp, jobs_rsp = self._client._send_batch(req)
+        batch_rsp = self._client._send_batch(req)
         batch_id = batch_rsp["id"]
-        if wait or fetch_results:
+        if wait:
             while batch_rsp["status"] in ["PENDING", "RUNNING"]:
                 time.sleep(RESULT_POLLING_INTERVAL)
-                batch_rsp, jobs_rsp = self._client._get_batch(batch_id)
+                batch_rsp = self._client._get_batch(batch_id)
 
-            if fetch_results:
-                batch_rsp, jobs_rsp = self._client._get_batch(
-                    batch_id, fetch_results=True
-                )
-        batch = Batch(**batch_rsp, jobs=jobs_rsp, _client=self._client)
```

Comment:
I'm now a bit confused. 
I'm pretty sure I added this code, in the last 6 months(?), specifically because this didn't happen

Reply:
Also note:
https://github.com/pasqal-io/pasqal-cloud/issues/66

Reply:
Tested it in dev and it does happen. Wait already checks that the status is complete. Once it's complete the data from the jobs, including results is already part of the batch payload.

Reply:
Of course we'll wait for the e2e pipeline to run successfully.

Reply:
Looking at the dgx_connector code, results are sent prior to updating batch status, so I don't see in what scenario a fetched completed batch (which already contains the results) would not have the most up to date results.

Reply:
Could you also do a manual safety check and ensure this doesn't result in a race condition? 

Also, set the Polling interval to something small, send say a few hundred batches and just verify that nothing goes wrong.
I'm just a bit skeptical because I remember when this code was introduced, and it was after we moved to FastAPI.

It could be that it was some faulty logic in core that has later been fixed. This behavior that this MR expects is of course what is the most sensible

Reply:
I'll run the check with a couple (hundred?) batches in parallel.
Maybe it was before we fixed some issues with async sqlalchemy sessions.
But  even then since the order of operations is as follows:
- send result for all jobs
- check this is the last job
- set batch status to complete
I'm unsure how a complete batch could have non up-to-date results

Reply:
@awennersteen following our conversation, if you still have doubts regarding the viability of this approach, would you mind pulling from this branch and running your usual tests battery? 
Otherwise can you resolve?

Reply:
it's fine for me